versioned/storage/cassandra/src/main/java/org/projectnessie/versioned/storage/cassandra/CassandraPersist.java

```diff
@@ -391,8 +400,13 @@ public boolean updateConditional(@Nonnull UpdateableObj expected, @Nonnull Updat
             .setString(COL_OBJ_ID.name(), serializeObjId(id))
             .setString(COL_OBJ_TYPE.name() + EXPECTED_SUFFIX, type.name())
             .setString(COL_OBJ_VERS.name() + EXPECTED_SUFFIX, expectedVersion)
-            .setString(COL_OBJ_VERS.name(), newVersion)
-            .setLong(COL_OBJ_REFERENCED.name(), referenced);
+            .setString(COL_OBJ_VERS.name(), newVersion);
+    if (newValue.referenced() != -1L) {
+      // -1 is a sentinel for AbstractBasePersistTests.deleteWithReferenced()
+      stmt = stmt.setLong(COL_OBJ_REFERENCED.name(), referenced);
+    } else {
+      stmt = stmt.setToNull(COL_OBJ_REFERENCED.name());
```

Comment:
Is saving an object with `referenced() == -1` a valid scenario?

Reply:
I hope not - that would be the end of world (long overflow from Long.MAX_VALUE ... becoming negative ... then reaching -1).

Reply:
But seriously - no, it's always a (positive) timestamp.

Reply:
So, why no throw in this `else` block then? I do not think we have to support reverting to old data in Nessie tables after upgrading to a new version. WDYT?

Reply:
It's the sentinel needed for tests - that's why the handling of -1 is in all `Persist` implementations

Reply:
It's not a valid production scenario though

Reply:
Good point (I missed the tests :facepalm: )

Reply:
No great choice there - either clutter the code with a weird configuration option (nooooo) or this weird -1 handling when writing.